File: LeetCode/ComplexNumberMultiplication.cpp
class Solution {
public:
    string complexNumberMultiply(string a, string b) {
        int i_plus = a.find('+',0);
        int ar = stoi(a.substr(0, i_plus), nullptr);
        int ai = stoi(a.substr(i_plus+1), nullptr); // stoi should stop at the i, not throw an error
        i_plus = b.find('+',0);
        int br = stoi(b.substr(0, i_plus), nullptr);
        int bi = stoi(b.substr(i_plus+1), nullptr);
        int cr = (ar * br) - (ai * bi);
        int ci = (ar * bi) + (ai * br);
        return to_string(cr) + "+" + to_string(ci) + "i";
    }
};

File: LeetCode/ComplexNumberMultiplication.cs
public class Solution
{
    public string ComplexNumberMultiply(string a, string b)
    {
        int i_plus = a.IndexOf('+');
        int ar = int.Parse(a.Substring(0, i_plus));
        int ai = int.Parse(a.Substring(i_plus + 1, a.Length - i_plus - 2)); // stoi should stop at the i, not throw an error
        i_plus = b.IndexOf('+');
        int br = int.Parse(b.Substring(0, i_plus));
        int bi = int.Parse(b.Substring(i_plus + 1, b.Length - i_plus - 2));
        int cr = (ar * br) - (ai * bi);
        int ci = (ar * bi) + (ai * br);
        return cr + "+" + ci + "i";
    }
};
